Utilities/rANS/include/rANS/internal/SymbolTable.h

```diff
@@ -0,0 +1,104 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   SymbolTable.h
+/// @author Michael Lettrich
+/// @since  2019-06-21
+/// @brief  Container for information needed to encode/decode a symbol of the alphabet
+
+#ifndef RANS_INTERNAL_SYMBOLTABLE_H
+#define RANS_INTERNAL_SYMBOLTABLE_H
+
+#include <vector>
+#include <cstdint>
+
+#include <fairlogger/Logger.h>
+
+#include "SymbolStatistics.h"
+
+namespace o2
+{
+namespace rans
+{
+namespace internal
+{
+
+template <typename T>
+class SymbolTable
+{
+
+ public:
+  SymbolTable(const SymbolStatistics& symbolStats, size_t probabiltyBits);
+
+  const T& operator[](int64_t index) const;
+
+  const T& getLiteralSymbol() const;
+
+  bool isLiteralSymbol(int64_t index) const;
+
+ private:
+  int mMin;
+  std::vector<T*> mIndex;
+  std::vector<T> mSymbols;
+};
+
+template <typename T>
+SymbolTable<T>::SymbolTable(const SymbolStatistics& symbolStats, size_t probabiltyBits) : mMin(symbolStats.getMinSymbol()), mIndex(), mSymbols()
+{
+  LOG(trace) << "start building symbol table";
+
+  mIndex.reserve(symbolStats.size());
+  mSymbols.reserve(symbolStats.size());
+
+  for (const auto& entry : symbolStats) {
+    const auto [symFrequency, symCumulated] = entry;
+    mSymbols.emplace_back(symCumulated, symFrequency, probabiltyBits);
+    mIndex.emplace_back(&mSymbols.back());
```

Comment:
What is the reason to build a vector of symbols and a vector of pointers to them?

Reply:
Thanks for the review. Oh, currently this does not do anything. There is potential to reduce the memory footprint of the SymbolTable on average by a factor of 4 and make it more cache efficient with one indirection which barely costs performance. This is what I started doing but had no time to finish it yet. Apparently I comited this code in there by accident.

Reply:
thanks, so, do you want to remove it or I merge this as it is now? For me it does not matter since this will not affect dictionary extraction

Reply:
Let's merge it.  I can polish this in a future PR.